File: 14_LongestCommonPrefix/14_v1.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        
        for i, c in enumerate(strs[0]):
            for str in strs:
                if len(str) <= i or str[i] != c:
                    return strs[0][0:i]
        
        return strs[0]
            

File: 14_LongestCommonPrefix/14_v1.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0)
            return "";
        
        for (int i = 0; strs[0].size(); i++){
            char c = strs[0][i];
            for (int j = 1; j < strs.size(); j++){
                if (strs[j].size() <= i || c != strs[j][i])
                    return string(strs[0].begin(), strs[0].begin() + i);
            }
        }
        
        return strs[0];
    }
};
